### ISSUE TITLE
Use new support for muting topics by stream ID when available

### ISSUE THREAD

**Issue by timabbott** [zulip/zulip-mobile#3244]
Related to https://github.com/zulip/zulip/issues/11126 and https://github.com/zulip/zulip/issues/11136, we now (in master) support sending a `stream_id` in the API for muting/unmuting topics.  We can't take advantage of this in mobile yet since no clients have that code, but I figured it's worth opening the issue for migrating.  

I expect several similar changes to be integrated in the server codebase in the coming weeks to support IDs in the API where possible.  We'll want to come up with a reasonable general strategy for detecting this sort of capability; possibly that's just a server version number check in the app codebase.  No urgency to migrate, in any case.

**Comment by borisyankov** [comment]
I think the optimal way is to do that once the server supports 'everything by id' and provide this as a value on `server_settings`.

The actual benefit to clients (including mobile) is once we can solely rely on IDs. 

One of the major things would be narrowing by IDs. Considering this might be a different format, we might include @gnprice suggestion for more 'readable' narrow format that I remember we discussed with you too, Tim (not 100% sure about that part)

**Comment by timabbott** [comment]
Yeah, that's a reasonable plan.  Narrowing by IDs is the main one that hasn't been started, though I think @showell has talked about doing it a few times.  We're probably merging "Send messages to PM recipients by IDs" in the next week or two.

**Comment by timabbott** [comment]
Just a small update that I'm adjusting the API documentation in https://github.com/zulip/zulip/issues/11136 to note that the old format is deprecated.  Presumably we can't remove it until some time after mobile adds support for it, though.

**Comment by chrisbobbe** [comment]
It looks like support for using stream IDs was https://github.com/zulip/zulip/commit/c35afb942, released in 2.0.0. There's been a [mention](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/compatibility.20documentation/near/1174950) that as of March, dropping support for 1.9.x and older might affect 2-4% of users.

Are we OK to switch completely over to using stream IDs for this feature, or should we check the server version and only use stream IDs if the server is at least 2.0.0?

**Comment by gnprice** [comment]
Looks like this was accidentally closed by #4635 -- the original revision of that PR would have resolved this, but in the end it didn't, because https://github.com/zulip/zulip-mobile/pull/4635#discussion_r652455585 .

I'll take care of this in an upcoming followup to #5192, as part of relying on server 2.0+.
